src/standalone_async/hello_async.cpp

```diff
@@ -0,0 +1,137 @@
+#include "hello_async.hpp"
+
+#include <exception>
+#include <stdexcept>
+#include <iostream>
+#include <map>
+
+namespace standalone_async {
+
+  /*
+  * This is an internal function used to return callback error messages instead of
+  * throwing errors.
+  * Usage:
+  *
+  * v8::Local<v8::Function> callback;
+  * CallbackError("error message", callback);
+  * return; // this is important to prevent duplicate callbacks from being fired!
+  */
+  inline void CallbackError(std::string message, v8::Local<v8::Function> callback) {
+  	v8::Local<v8::Value> argv[1] = { Nan::Error(message.c_str()) };
+  	Nan::MakeCallback(Nan::GetCurrentContext()->Global(), callback, 1, argv);
+  }
+
+  // expensive allocation of std::map, querying, and string comparison
+  std::string do_expensive_work(bool louder, std::size_t work_to_do=100000) {  
+
+      std::map<std::size_t,std::string> container;  
+
+      for (std::size_t i=0;i<work_to_do;++i) {
+          container.emplace(i,std::to_string(i));
+      }  
+
+      for (std::size_t i=0;i<work_to_do;++i) {
+          std::string const& item = container[i];
+          if (item != std::to_string(i)) {
+              abort(); // should never get here
```

Comment:
@daniel-j-h great point. I [replaced it with a Callback](https://github.com/mapbox/node-cpp-skel/pull/46/commits/ec533587ef18d79647eb0fa280f4d372b4af923f#diff-881a9b2c4458d0d79b1d20ac96bba548R38), now troubleshooting compile errors.